server/fleet/datastore.go

```diff
@@ -1551,6 +1551,9 @@ type Datastore interface {
 
 	// BatchSetSoftwareInstallers sets the software installers for the given team or no team.
 	BatchSetSoftwareInstallers(ctx context.Context, tmID *uint, installers []*UploadSoftwareInstallerPayload) error
+
+	// HasSelfServiceSoftwareInstallers returns true if self-service software installers are available for the team or globally.
+	HasSelfServiceSoftwareInstallers(ctx context.Context, platform string, teamID *uint) (bool, error)
```

Comment:
See comment regarding the specification that software installers must be associated with a team (i.e. they are excluded from presentation at the global "All teams" level).